clang/lib/Driver/Driver.cpp

```diff
@@ -5105,6 +5105,62 @@ class OffloadingActionBuilder final {
       }
     }
 
+    // Return whether to use native bfloat16 library.
+    bool selectBfloatLibs(const ToolChain *TC, bool &useNative) {
+      const OptTable &Opts = C.getDriver().getOpts();
+      const char *TargetOpt = nullptr;
+      const char *DeviceOpt = nullptr;
+      bool needLibs = false;
+      for (auto *A : Args) {
+        llvm::Triple *TargetBE = nullptr;
+
+        auto GetTripleIt = [&, this](llvm::StringRef Triple) {
+          llvm::Triple TargetTriple{Triple};
+          auto TripleIt = llvm::find_if(SYCLTripleList, [&](auto &SYCLTriple) {
+            return SYCLTriple == TargetTriple;
+          });
+          return TripleIt != SYCLTripleList.end() ? &*TripleIt : nullptr;
+        };
+
+        if (A->getOption().matches(options::OPT_fsycl_targets_EQ)) {
+          // When a generic target "spir64" is used with other AOT targets
+          // we use fallback libraries.
+          if (TC->getTriple().getSubArch() == llvm::Triple::NoSubArch)
+            needLibs = DeviceLinkerInputs.size() > 1;
+          else
+            needLibs = true;
```

Comment:
I tried a simple test case with (1), and it seemed to run.  Is the assertion you hit somehow triggered by your change, or is it an existing bug that we didn't hit before because this usage is not well tested?  Obviously, we shouldn't fix a general problem with multi-target compilation in this PR.

Actually, I'm wondering if my understanding is incorrect.  In case (1), does the DPC++ compiler generate one device image or two?  I was originally thinking it generated two, but I'm beginning to wonder if that's the case.